lib/core/Proxy.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const WebSocket = require('ws');
+const { config, kCleanup, kAddNewContext } = require('../config/constants');
+const logger = require('../util/loggerFactory');
+const Context = require('./Context');
+const {
+  isReconnectHeader,
+  extractConnectionId,
+  extractReconnectId,
+} = require('../util/util');
+const { isString } = require('../util/typeSanity');
+const Instrumentation = require('../util/Instrumentation');
+const http = require('http');
+const { URL } = require('url');
+
+/**
+ * Proxy is the entrypoint and instantiates the context among the socket connection.
+ * WebSocket server is created & maintaining contexts of connections that are established.
+ *
+ * Registers the contexts holding all the connections which are created.
+ */
+class Proxy {
+  /**
+   * Creates the Proxy.
+   */
+  constructor() {
+    this.httpServer = http.createServer(this.requestHandler.bind(this));
+    this.httpServer.listen(config.port);
```

Comment:
Not bound to a particular host, pass the second argument is the host where it needs to be listening. Set the second argument as the `config.hostname`.

Reply:
fixed